src/Repository/Storable.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Friendica\Repository;
+
+use Friendica\Collection;
+use Friendica\Model;
+use Friendica\Repository;
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Network\HTTPException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Storable repositories are linked to a main table with an id primary key and will instanciate Models and Collections
+ * related to that table.
+ */
+abstract class Storable extends Repository
+{
+	const LIMIT = 30;
+
+	/** @var Database */
+	protected $dba;
+
+	/** @var string */
+	protected static $table_name;
+
+	/** @var Model */
+	protected static $model_class;
+
+	/** @var Collection */
+	protected static $collection_class;
+
+	public function __construct(Database $dba, LoggerInterface $logger)
+	{
+		parent::__construct($logger);
+
+		$this->dba = $dba;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * Fetches a single model record. The condition array is expected to contain a unique index (primary or otherwise).
+	 *
+	 * Chainable.
+	 *
+	 * @param array $condition
+	 * @return Model
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function selectFirst(array $condition)
+	{
+		$data = $this->dba->selectFirst(static::$table_name, [], $condition);
+
+		if (!$data) {
+			throw new HTTPException\NotFoundException(static::class . ' record not found.');
+		}
+
+		return new static::$model_class($data);
+	}
+
+	/**
+	 * Populates a Collection according to the condition.
+	 *
+	 * Chainable.
+	 *
+	 * @param array $condition
+	 * @param array $params
+	 * @return Collection
+	 * @throws \Exception
+	 */
+	public function select(array $condition = [], array $params = [])
+	{
+		$result = $this->dba->select(static::$table_name, [], $condition, $params);
+
+		$models = [];
+
+		while ($record = $this->dba->fetch($result)) {
+			$models[] = new static::$model_class($this->dba, $this->logger, $record);
+		}
+
+		return new static::$collection_class($models);
+	}
+
+	/**
+	 * Populates the collection according to the condition. Retrieves a limited subset of models depending on the boundaries
+	 * and the limit. The total count of rows matching the condition is stored in the collection.
+	 *
+	 * Chainable.
+	 *
+	 * @param array $condition
+	 * @param array $params
+	 * @param int?  $max_id
+	 * @param int?  $since_id
+	 * @param int   $limit
+	 * @return $this
```

Comment:
Really `$this` ? I guess this one is where I'm stuck too .. It's `Model[]` or a `Collection` class, but depending on the final repository, it can be a `Introduction[]` array or an `Introductions` class ;-)

Reply:
It's a remnant from the previous Collection code that was self-fetching, so methods were chainable.